crypto/serializer/deserializer_lib.c

```diff
@@ -37,10 +37,11 @@ int OSSL_DESERIALIZER_from_bio(OSSL_DESERIALIZER_CTX *ctx, BIO *in)
 
     ok = deser_process(NULL, &data);
 
-    /* Clear any cached passphrase */
-    OPENSSL_clear_free(ctx->cached_passphrase, ctx->cached_passphrase_len);
-    ctx->cached_passphrase = NULL;
-    ctx->cached_passphrase_len = 0;
+    /* Clear any internally cached passphrase */
+    if (!ctx->flag_user_passphrase) {
```

Comment:
Maybe my brain isnt working - is this supposed to be using ! ?

Reply:
Not even sure the if i s required at all.

Reply:
Yes.  If the flag is set, then the passphrase was provided with `OSSL_DESERIALIZER_CTX_set_passphrase()`, i.e. not cached internally and is therefore the responsibility of the caller.  We should NOT spring on the caller the surprise of stealth freeing what they gave us.